federation/pkg/federation-controller/service/dns.go

```diff
@@ -16,25 +16,230 @@ limitations under the License.
 
 package service
 
+import (
+	"fmt"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+const (
+	// minDnsTtl is the minimum safe DNS TTL value to use (in seconds).  Less than this, and some caching DNS servers might ignore it.
+	minDnsTtl = 180
+)
+
+// getHealthyEndpoints returns the hostnames and/or IP addresses of healthy endpoints for the service, at a zone, region and global level (or an error)
+func (s *ServiceController) getHealthyEndpoints(clusterName string, cachedService *cachedService) (zoneEndpoints, regionEndpoints, globalEndpoints []string, err error) {
```

Comment:
Remove `err` in `err error`. It is not necessary to use a named return. You are not using that feature in your method anyway.